articles/115_topic_and_service_name_mapping.md

```diff
@@ -0,0 +1,313 @@
+On systems where the implementation is handled for us by DDS (like with Connext), we use `rS_` as the common prefix.
+
+Here is a non-exhaustive list of prefixes:
+
+| ROS Subsystem        | Prefix |
+|----------------------|--------|
+| ROS Topics           | rt_    |
+| ROS Service Request  | rq_    |
+| ROS Service Response | rs_    |
+| ROS Service          | rS_    |
+| ROS Parameter        | rp_    |
+| ROS Action           | ra_    |
+
+### Substitution of the Namespace Delimiter
+
+The namespace delimiter in ROS 2 topic and service names, a forward slash (`/`), will be replaced with double underscores (`__`).
+The leading forward slash (`/`) is not replaced with double underscores (`__`), and is replaced instead by the ROS specific name prefix.
+
+### ROS to DDS Name Conversion Examples
+
+Here are some examples of translations between ROS topic names and DDS topic names:
+
+| ROS Name          | Subsystem | DDS Name              |
+|-------------------|-----------|-----------------------|
+| `/foo`            | Topic     | `rt_foo`              |
+| `/foo/bar`        | Topic     | `rt_foo__bar`         |
+| `/_foo/bar`       | Topic     | `rt__foo__bar`        |
+| `/foo/_bar`       | Topic     | `rt_foo___bar`        |
+| `/_foo/_bar`      | Topic     | `rt__foo___bar`       |
+| `/_foo/_bar/_baz` | Topic     | `rt__foo___bar___baz` |
+| `/foo/_/bar`      | Topic     | `rt_foo_____bar`      |
+
+## Concerns/Alternatives
+
+This section lists concerns about the proposed design and alternatives that were considered.
+
+### Alternative ROS to DDS Mapping
+
+There was some discussion of alternatives and concerns with respect to the ROS -> DDS translation.
+
+#### Capital Letter Substitution Alternative
+
+Since the forward slash (`/`) is replaced with double underscores (`__`) when translating to DDS from ROS topic names, two characters out of the 256 character limit are lost with each additional namespace.
+One proposed alternative was to add a constraint that ROS topic names could not use capital letters, and then capital letters could be used as the stand in for the forward slashes (`/`).
+
+Trade-offs:
+
+- Uses one less character per namespace and makes it easier to calculate the maximum length of a ROS topic or service name.
+- Prevents users from using capital letters in their names, which is constraining and might be a problem for backwards compatibility with ROS 1 topics and services.
+
+Rational:
+
+Preventing users from using capital letters was too constraining for the added benefit.
+
+#### Prefix with Double Underscores
+
+This alternative differs only in that it uses two underscores in the prefix, i.e. `rt__` rather than `rt_`.
+
+Trade-offs:
+
+- More consistent with replacement of other forward slashes (`/`)
+- Uses one more character
+
+Rational:
+
+Slight preference given to the shorter alternative.
```

Comment:
Well then we'll have to vote on it, because I disagree. We don't have any slack in the limitation on the name length, so any extra work we can do, which doesn't also restrict the user, to maximize the available characters is worth it in my opinion. This isn't a complex rule and in all likelihood would be a single line in any language where it is implemented.